test/images/vulnerability-scan/scan-image.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+
+VULNERABILITIES_EXIST=1
+
+get_digest() {
+  #number of retries=3
+  for _ in {1..3}
+  do
+    #Get the digest of the image
+    digest_http_code=$(curl -s -w '%{http_code}' -o /tmp/digest.json -H "Accept: application/vnd.quay.distribution.manifest.v2+json Content-type: application/json Authorization: Bearer $AUTH_BEARER_TOKEN" https://quay.io/api/v1/repository/redhat-pipeline-service/"$IMAGE_NAME")
+
+    if [[ "$digest_http_code" == "200" ]]; then
+      digest=$(jq '.tags[] | select(.name == "main").manifest_digest ' /tmp/digest.json | tr -d '"')
+      rm -f /tmp/digest.json
+      get_vulnerabilities
+      break
+    else
+      printf "Error while fetching digests from Quay. Status code: %s\n" "${digest_http_code}"
+    fi
```

Comment:
What do you think about adding a sleep before retry?

Reply:
I thought the default time curl takes and retires to try and fetch was enough. But yeah, don't mind adding more time to reduce the chance of failure. 
Added sleep 30.

Reply:
there could be some improvement here to avoid unnecessary waits, something like:
~~~
for i in {1..3}
...
else
  printf "Error while fetching digests from Quay. Status code: %s\n" "${digest_http_code}"
  if i < 3
    sleep 30
  fi
~~~

Reply:
Done. Improved the log message as well.